source/common/http/conn_manager_impl.cc

```diff
@@ -173,6 +173,9 @@ StreamDecoder& ConnectionManagerImpl::newStream(StreamEncoder& response_encoder)
   new_stream->response_encoder_ = &response_encoder;
   new_stream->response_encoder_->getStream().addCallbacks(*new_stream);
   config_.filterFactory().createFilterChain(*new_stream);
+  if (underlying_connection_above_high_watermark_) {
```

Comment:
Ah I see. OK. Can you add comments here? I think in general more comments would be useful in cases like this.